packages/components/src/spectrum/utils/itemUtils.ts

```diff
@@ -137,9 +137,10 @@ export function getItemKey<
  */
 export function getItemTextValue<T>(item: ItemElement<T>): string | undefined {
   if (item.props.textValue == null) {
+    const itemKey = item.key !== undefined ? String(item.key) : undefined;
```

Comment:
I would use the looser == null check and check for equality instead of inequality. As a pattern, negation checks are slightly harder to read since they require an extra boolean flip in your brain

```typescript
const itemKeyStr = item.key == null ? undefined : String(item.key)`
```

Reply:
Regarding your Slack question about `ITEM_EMPTY_STRING_TEXT_VALUE`, you could incorporate that in `getItemTextValue` so that it also gets applied to the case where it falls back to key, if key were `''`. 

I would leave the `useRenderNormalizedItem` as-is in regards to `ITEM_EMPTY_STRING_TEXT_VALUE`. It's slightly different in that you are dealing with a `NormalizedItem` instead of `ItemElement`, so I think you'd have to do more refactoring that probably isn't necessary